### DISCUSSION TITLE
Unknown message type

### DISCUSSION THREAD

**Discussion by antonyramsy** [foxglove/community#19] [Q&A] [answered]
I'm using foxglove to replay my ros2 bag (db3) file. although most of the topics are displayed without any problem, there are certain topics whose message type can't be found and then the whole replaying process is terminated with an error.

details:
my db3 file mostly contains Mavros topics but foxglove can't display for example /mavros/state and /mavros/setpoint_raw/target_global which both have mavros_msgs/msg/... types. 

the error message:
```
Error: Unknown message type: mavros_msgs/msg/GlobalPositionTarget
    at Rosbag2.decodeMessage [as decoder] (file:_____/foxglove-studio/resources/app.asar/renderer/1950.js:5:4463)
    at MessageIterator.next (file:_____/foxglove-studio/resources/app.asar/renderer/1950.js:5:1932)
    at async RosDb3IterableSource.messageIterator (file:_____/foxglove-studio/resources/app.asar/renderer/6033.js:1:3581)
```

however, all other topics which also have their unique and non-standard types has warning but are displayed without any problem.
warning message example:
```
ROS 2 .db3 files do not contain message definitions, so only well-known ROS types are supported in Foxglove Studio. As a workaround, you can convert the db3 file to mcap using the mcap CLI. For more information, see: https://foxglove.dev/docs/studio/connection/local-file
```

 i can't figure out the source of the problem. is there anything I can do to resolve the issue with the Unknown message type?

**Comment by defunctzombie** [comment, marked as answer]
We recommend converting your DB3 files into `mcap` files (https://foxglove.dev/docs/studio/connection/ros2#files). This will put your custom message definitions into the `mcap` file and you will be able to load this in studio.
